ql/src/java/org/apache/hadoop/hive/ql/ddl/table/partition/show/ShowPartitionsOperation.java

```diff
@@ -116,4 +120,22 @@ private List<String> getPartitionNames(Table tbl) throws HiveException {
         desc.getOrder(), desc.getLimit());
     return partNames;
   }
+
+  private boolean isPartitionedIcebergTable(Table tbl) {
+    String t = tbl.getParameters().get("table_type");
+    return t != null && t.equalsIgnoreCase("ICEBERG") &&
```

Comment:
What about `"ICEBERG".equalsIgnoreCase(tbl.getParameters().get("table_type"))` ?

Reply:
Also, I'm thinking (not in this ticket) but is it worth adding an extra method to Table to get parameter by name? 
E.g: getParameter(String name)

<img width="1519" alt="image" src="https://user-images.githubusercontent.com/1486749/216643120-c99f7715-a643-47ed-ace7-885fc18825fa.png">